src/Plugin/views/access/GroupPermission.php

```diff
@@ -0,0 +1,234 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $usesOptions = TRUE;
+
+  /**
+   * The permission manager.
+   *
+   * @var \Drupal\og\PermissionManagerInterface
+   */
+  protected $permissionManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The bundle information service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  protected $bundleInfo;
+
+  /**
+   * The OG access service.
+   *
+   * @var \Drupal\og\OgAccess
+   */
+  protected $ogAccess;
+
+  /**
+   * The OG context provider.
+   *
+   * @var \Drupal\Core\Plugin\Context\ContextProviderInterface
+   */
+  protected $ogContext;
+
+  /**
+   * The group manager.
+   *
+   * @var \Drupal\og\GroupTypeManager
+   */
+  protected $groupTypeManager;
+
+  /**
+   * Constructs a Permission object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\og\PermissionManagerInterface $permission_manager
+   *   The permission handler.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $og_context
+   *   The OG context provider.
+   * @param \Drupal\og\GroupTypeManager $group_type_manager
+   *   The group manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, PermissionManagerInterface $permission_manager, EntityTypeManagerInterface $entity_type_manager, OgAccessInterface $og_access, ContextProviderInterface $og_context, GroupTypeManager $group_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->permissionManager = $permission_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->ogAccess = $og_access;
+    $this->ogContext = $og_context;
+    $this->groupTypeManager = $group_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.permission_manager'),
+      $container->get('entity_type.manager'),
+      $container->get('og.access'),
+      $container->get('og.context'),
+      $container->get('og.group_type_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(AccountInterface $account) {
+    $group = $this->getGroup();
+    if ($group instanceof ContentEntityInterface) {
```

Comment:
`::getGroup()` always returns either a `ContentEntityInterface` or `NULL`, so this can be simplified to:

```
if ($group) {
  // ...
}
```

That makes the intention clearer, now the code appears to hint that `$group` might be a config entity under some circumstances.